twitter/src/main/java/com/hazelcast/jet/contrib/twitter/TwitterSources.java

```diff
@@ -0,0 +1,375 @@
+            for (String item : buffer) {
+                sourceBuffer.add(item);
+            }
+            buffer.clear();
+        }
+
+        private void fillTimestampedBuffer(SourceBuilder.TimestampedSourceBuffer<String> sourceBuffer) {
+            queue.drainTo(buffer, MAX_FILL_ELEMENTS);
+            for (String item : buffer) {
+                JsonObject object = Json.parse(item).asObject();
+                long timestamp = Long.parseLong(object.getString("timestamp_ms", "0"));
+                sourceBuffer.add(item, timestamp);
+            }
+            buffer.clear();
+        }
+
+        private void close() {
+            if (client != null) {
+                client.stop();
+            }
+        }
+
+    }
+
+    /**
+     * Context for the batch source of Twitter's Search API
+     */
+    private static final class TwitterBatchSourceContext {
+        private QueryResult searchResult;
+        private final Twitter twitter4JClient;
+
+        /**
+         * @param credentials a Twitter OAuth1 credentials that consists "consumerKey",
+         *                    "consumerSecret", "token", "tokenSecret" keys.
+         * @param query       a search query
+         */
+        private TwitterBatchSourceContext(
+                @Nonnull Properties credentials,
+                @Nonnull String query
+        ) throws TwitterException {
+            checkTwitterCredentials(credentials);
+            String consumerKey = credentials.getProperty("consumerKey");
+            String consumerSecret = credentials.getProperty("consumerSecret");
+            String token = credentials.getProperty("token");
+            String tokenSecret = credentials.getProperty("tokenSecret");
+            ConfigurationBuilder cb = new ConfigurationBuilder();
+            cb.setDebugEnabled(true)
+                    .setOAuthConsumerKey(consumerKey)
+                    .setOAuthConsumerSecret(consumerSecret)
+                    .setOAuthAccessToken(token)
+                    .setOAuthAccessTokenSecret(tokenSecret);
+            this.twitter4JClient = new TwitterFactory(cb.build())
+                    .getInstance();
+            this.searchResult = twitter4JClient.search(new Query(query));
+        }
+
+        private void fillBuffer(SourceBuilder.SourceBuffer<Status> sourceBuffer) throws TwitterException {
+            try {
+                List<Status> tweets = searchResult.getTweets();
+                for (Status tweet : tweets) {
+                    sourceBuffer.add(tweet);
+                }
+                searchResult = twitter4JClient.search(searchResult.nextQuery());
+            } catch (TwitterException te) {
+                te.printStackTrace();
+                System.out.println("Failed to search tweets: " + te.getMessage());
+            }
+        }
+        private void close() {
+            // TODO: I don't know what to do in here. Twitter4J client does not have any destructor.
```

Comment:
Doesn't `Twitter` client have the `stop()` method ?

Reply:
this client has not the stop() and any stop() like method.

Reply:
Then you can remove this method and remove also the destroy function from the SourceBuilder